nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/DiagnosticProperties.java

```diff
@@ -0,0 +1,125 @@
+
+    private static final String MAX_FILE_COUNT_PROP_NAME = "nifi.diag.filecount.max";
+    private static final int MAX_FILE_COUNT_DEFAULT_VALUE = Integer.MAX_VALUE;
+
+    private static final String MAX_SIZE_PROP_NAME = "nifi.diag.size.max.byte";
+    private static final int MAX_SIZE_DEFAULT_VALUE = Integer.MAX_VALUE;
+
+    private static final String VERBOSE_PROP_NAME = "nifi.diag.verbose";
+    private static final boolean VERBOSE_DEFAULT_VALUE = false;
+
+    private final String dirPath;
+    private final int maxFileCount;
+    private final int maxSizeInBytes;
+    private final boolean verbose;
+    private final boolean allowed;
+
+    public DiagnosticProperties() throws IOException {
+        BootstrapProperties properties = NiFiBootstrapUtils.loadBootstrapProperties();
+        this.dirPath = properties.getProperty(DIR_PROP_NAME, DIR_DEFAULT_VALUE);
+        this.maxFileCount = getPropertyAsInt(properties.getProperty(MAX_FILE_COUNT_PROP_NAME), MAX_FILE_COUNT_DEFAULT_VALUE);
+        this.maxSizeInBytes = getPropertyAsInt(properties.getProperty(MAX_SIZE_PROP_NAME), MAX_SIZE_DEFAULT_VALUE);
+        this.verbose = getPropertyAsBoolean(properties.getProperty(VERBOSE_PROP_NAME), VERBOSE_DEFAULT_VALUE);
+        this.allowed = getPropertyAsBoolean(properties.getProperty(ALLOWED_PROP_NAME), ALLOWED_DEFAULT_VALUE);
+        createDiagDir();
+    }
+
+    public Path getOldestFile() throws IOException {
+        Comparator<? super Path> lastModifiedComparator = Comparator.comparingLong(p -> p.toFile().lastModified());
+
+        final List<Path> oldestFiles;
+
+        try (Stream<Path> paths = Files.walk(Paths.get(dirPath))) {
+            oldestFiles = paths.filter(Files::isRegularFile)
+                    .sorted(lastModifiedComparator)
+                    .limit(1)
+                    .collect(Collectors.toList());
+        }
+
+        return oldestFiles.get(0);
+    }
+
+    public boolean isFileCountExceeded() {
+        return new File(dirPath).list().length >= maxFileCount;
+    }
+
+    public boolean isSizeExceeded() {
+        return FileUtils.getDirectorySize(Paths.get(dirPath), logger) >= maxSizeInBytes;
+    }
+
+    public boolean isVerbose() {
+        return verbose;
+    }
+
+    public String getDirPath() {
+        return dirPath;
+    }
+
+    public boolean isAllowed() {
+        return allowed;
+    }
+
+    private static int getPropertyAsInt(final String property, final int defaultValue) {
+        try {
+            return Integer.parseInt(property);
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
+
+    private static boolean getPropertyAsBoolean(final String property, final boolean defaultValue) {
+        try {
+            return Boolean.parseBoolean(property);
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
+
+    private void createDiagDir() {
```

Comment:
Is an abbreviation that only save a couple characters worth it?
```suggestion
    private void createDiagnosticsDir() {
```